### PR TITLE
All 15 contribution events handled

### PR DESCRIPTION
**Pull request by hero101**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: src/services/external/elasticsearch/types/contribution.type.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  OPPORTUNITY_CONTENT_EDITED: 'OPPORTUNITY_CONTENT_EDITED',
+  OPPORTUNITY_JOINED: 'OPPORTUNITY_JOINED',
+  CALLOUT_CREATED: 'CALLOUT_CREATED',
+  CALLOUT_COMMENT_CREATED: 'CALLOUT_COMMENT_CREATED',
+  CALLOUT_CARD_CREATED: 'CALLOUT_CARD_CREATED',
+  CALLOUT_CANVAS_CREATED: 'CALLOUT_CANVAS_CREATED',
+  CALLOUT_CARD_COMMENT_CREATED: 'CALLOUT_CARD_COMMENT_CREATED',
+  CALLOUT_CANVAS_EDITED: 'CALLOUT_CANVAS_EDITED',
+  UPDATE_CREATED: 'UPDATE_CREATED',
+} as const;
```
Comment (valentinyanakiev) [inline src/services/external/elasticsearch/types/contribution.type.ts:17]: Do you need that cast?

File: src/domain/community/community/community.resolver.mutations.ts [hunk 2]
```diff
@@ -400,6 +403,42 @@ export class CommunityResolverMutations {
       CommunityRole.MEMBER,
       agentInfo
     );
+
+    if (community.type === CommunityType.HUB) {
```
Comment (valentinyanakiev) [inline src/domain/community/community/community.resolver.mutations.ts]: A `switch` would be much more readable here.

File: src/domain/challenge/hub/hub.service.ts [hunk 3]
```diff
@@ -651,7 +651,7 @@ export class HubService {
 
   async createChallengeInHub(
     challengeData: CreateChallengeOnHubInput,
-    agentInfo?: AgentInfo
+    agentInfo: AgentInfo
```
Comment (valentinyanakiev) [inline src/domain/challenge/hub/hub.service.ts]: I am not sure why AgentInfo is optional here, but this change can potentially cause a regression.

File: src/domain/challenge/challenge/challenge.service.ts [hunk 4]
```diff
@@ -82,7 +84,7 @@ export class ChallengeService {
   async createChallenge(
     challengeData: CreateChallengeInput,
     hubID: string,
-    agentInfo?: AgentInfo
+    agentInfo: AgentInfo
```
Comment (valentinyanakiev) [inline src/domain/challenge/challenge/challenge.service.ts]: I am not sure why AgentInfo is optional here, but this change can potentially cause a regression.

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]

File: src/domain/challenge/challenge/challenge.service.ts (hunk 4 above)
Comment (hero101) [reply src/domain/challenge/challenge/challenge.service.ts]: Reverted, it was for an earlier version of the code.

File: src/domain/challenge/hub/hub.service.ts (hunk 3 above)
Comment (hero101) [reply src/domain/challenge/hub/hub.service.ts]: Reverted.

File: src/services/external/elasticsearch/types/contribution.type.ts (hunk 1 above)
Comment (hero101) [reply src/services/external/elasticsearch/types/contribution.type.ts:17]: In this specific use case probably not - but generally I don't want `CONTRIBUTION_TYPE` to be changed further.

**Pull request merged**